### PR TITLE
[DROOLS-7164] Drools 8 docs : Rule Unit docs review and update

### PR DESCRIPTION
**Pull request by tkobayas**
**JIRA**: 
https://issues.redhat.com/browse/DROOLS-7164

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] tests</b>

- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

- <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

- <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-main</b>

- for <b>quarkus lts checks</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins run quarkus-lts</b>

- for a <b>specific quarkus lts check</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-lts</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] mandrel</b>

 - for <b>mandrel lts checks</b>  
  Run native checks against Mandrel image and quarkus lts branch
  Please add comment: <b>Jenkins run mandrel-lts</b>

- for a <b>specific mandrel lts check</b>  
  Run native checks against Mandrel image and quarkus lts branch
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] mandrel-lts</b>
</details>

### CODE REVIEW

**Comment by kie-ci3** [comment]
**(tests) - kogito-examples job** `#565` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-examples/565/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
[2022-09-22T05:49:53.241Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/1.27.x/nightly.quarkus-branch/kogito-all_ws-cleanup_1663825783202/kogito-apps/ui-packages/packages/patternfly-base/node_modules/@patternfly/react-tokens/dist/esm/c_file_upload_m_loading__file_details_before_Right.js’: No such file or directory<br/>[2022-09-22T05:49:53.241Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/1.27.x/nightly.quarkus-branch/kogito-all_ws-cleanup_1663825783202/kogito-apps/ui-packages/packages/patternfly-base/node_modules/@patternfly/react-tokens/dist/esm/c_form_GridGap.js’: No such file or directory<br/>[2022-09-22T05:49:53.241Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/1.27.x/nightly.quarkus-branch/kogito-all_ws-cleanup_1663825783202/kogito-apps/ui-packages/packages/patternfly-base/node_modules/@patternfly/react-tokens/dist/esm/c_form__actions_MarginBottom.js’: No such file or directory<br/>[2022-09-22T05:49:53.241Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/1.27.x/nightly.quarkus-branch/kogito-all_ws-cleanup_1663825783202/kogito-apps/ui-packages/packages/patternfly-base/node_modules/@patternfly/react-tokens/dist/esm/c_form__actions_MarginLeft.js’: No such file or directory<br/>[2022-09-22T05:49:53.241Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/1.27.x/nightly.quarkus-branch/kogito-all_ws-cleanup_1663825783202/kogito-apps/ui-packages/packages/patternfly-base/node_modules/@patternfly/react-tokens/dist/esm/c_form__actions_MarginRight.js’: No such file or directory<br/>[2022-09-22T05:49:53.241Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/1.27.x/nightly.quarkus-branch/kogito-all_ws-cleanup_1663825783202/kogito-apps/ui-packages/packages/patternfly-base/node_modules/@patternfly/react-tokens/dist/esm/c_form__actions_MarginTop.js’: No such file or directory<br/>[2022-09-22T05:49:53.497Z] 11G	/home/jenkins/workspace<br/>[2022-09-22T05:49:53.497Z] 0	/home/jenkins/.pki<br/>[2022-09-22T05:49:53.497Z] 0	/home/jenkins/caches<br/>[2022-09-22T05:49:54.057Z] 673M	/home/jenkins/.gradle<br/>[2022-09-22T05:49:54.057Z] 3.0M	/home/jenkins/.npm<br/>[2022-09-22T05:49:54.057Z] 0	/home/jenkins/.yarn<br/>[2022-09-22T05:49:54.057Z] 100M	/home/jenkins/tools<br/>[2022-09-22T05:49:54.057Z] 16G	/home/jenkins<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Install build-chain tool)<br/>Stage "Install build-chain tool" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build projects)<br/>Stage "Build projects" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Sonar analysis)<br/>Stage "Sonar analysis" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (check space after build)<br/>Stage "check space after build" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2022-09-22T05:49:54.972Z] + wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-examples/565/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Comment by kie-ci3** [comment]
**(tests) - kogito-apps job** `#556` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-apps/556/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
[2022-09-22T05:50:01.841Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/1.27.x/nightly.quarkus-branch/kogito-all_ws-cleanup_1663825783202/kogito-examples-events/serverless-workflow-examples/serverless-workflow-greeting-rpc-quarkus/serverless-workflow-greeting-client-rpc-quarkus/target/quarkus-app/lib/main/io.quarkus.quarkus-smallrye-stork-999-SNAPSHOT.jar’: No such file or directory<br/>[2022-09-22T05:50:01.841Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/1.27.x/nightly.quarkus-branch/kogito-all_ws-cleanup_1663825783202/kogito-examples-events/serverless-workflow-examples/serverless-workflow-greeting-rpc-quarkus/serverless-workflow-greeting-client-rpc-quarkus/target/quarkus-app/lib/main/io.smallrye.stork.stork-api-1.1.2.jar’: No such file or directory<br/>[2022-09-22T05:50:01.841Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/1.27.x/nightly.quarkus-branch/kogito-all_ws-cleanup_1663825783202/kogito-examples-events/serverless-workflow-examples/serverless-workflow-greeting-rpc-quarkus/serverless-workflow-greeting-client-rpc-quarkus/target/quarkus-app/lib/main/io.smallrye.stork.stork-core-1.1.2.jar’: No such file or directory<br/>[2022-09-22T05:50:01.842Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/1.27.x/nightly.quarkus-branch/kogito-all_ws-cleanup_1663825783202/kogito-examples-events/serverless-workflow-examples/serverless-workflow-greeting-rpc-quarkus/serverless-workflow-greeting-client-rpc-quarkus/target/quarkus-app/lib/main/io.quarkus.quarkus-grpc-stubs-999-SNAPSHOT.jar’: No such file or directory<br/>[2022-09-22T05:50:01.842Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/1.27.x/nightly.quarkus-branch/kogito-all_ws-cleanup_1663825783202/kogito-examples-events/serverless-workflow-examples/serverless-workflow-greeting-rpc-quarkus/serverless-workflow-greeting-client-rpc-quarkus/target/quarkus-app/lib/main/io.grpc.grpc-stub-1.48.1.jar’: No such file or directory<br/>[2022-09-22T05:50:01.842Z] du: cannot access ‘/home/jenkins/workspace/KIE/kogito/1.27.x/nightly.quarkus-branch/kogito-all_ws-cleanup_1663825783202/kogito-examples-events/serverless-workflow-examples/serverless-workflow-greeting-rpc-quarkus/serverless-workflow-greeting-client-rpc-quarkus/target/quarkus-app/lib/main/io.grpc.grpc-api-1.48.1.jar’: No such file or directory<br/>[2022-09-22T05:50:01.842Z] 1.6G	/home/jenkins/workspace<br/>[2022-09-22T05:50:01.842Z] 0	/home/jenkins/.pki<br/>[2022-09-22T05:50:01.842Z] 0	/home/jenkins/caches<br/>[2022-09-22T05:50:01.842Z] 673M	/home/jenkins/.gradle<br/>[2022-09-22T05:50:01.842Z] 3.0M	/home/jenkins/.npm<br/>[2022-09-22T05:50:01.842Z] 0	/home/jenkins/.yarn<br/>[2022-09-22T05:50:01.842Z] 100M	/home/jenkins/tools<br/>[2022-09-22T05:50:01.842Z] 6.6G	/home/jenkins<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Install build-chain tool)<br/>Stage "Install build-chain tool" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build projects)<br/>Stage "Build projects" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Sonar analysis)<br/>Stage "Sonar analysis" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (check space after build)<br/>Stage "check space after build" skipped due to earlier failure(s)<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Post Actions)<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2022-09-22T05:50:02.757Z] + wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-apps/556/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

File: drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_KIEBuilding-section.adoc [hunk 1]
```diff
@@ -313,11 +313,13 @@ In general this is not recommended, and the Maven plugin should always be used.
 [[_enginedependency]]
 == Engine dependency
 
-{PRODUCT} has 2 types of engine dependency that aggregates a required collection of dependencies.
+{PRODUCT} has 3 types of engine dependency that aggregates a required collection of dependencies.
```
Comment (kaldesai) [inline drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_KIEBuilding-section.adoc]:
```suggestion
{PRODUCT} has 3 types of engine dependency that aggregate a required collection of dependencies.
```

File: drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_KIEBuilding-section.adoc [hunk 2]
```diff
@@ -313,11 +313,13 @@ In general this is not recommended, and the Maven plugin should always be used.
...
 
-*`drools-engine`* is the standard engine that enables executable model.
+*`drools-ruleunits-engine`* is the standard engine for rule units that enables executable model.
```
Comment (kaldesai) [inline drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_KIEBuilding-section.adoc]:
```suggestion
*`drools-ruleunits-engine`* is the standard engine for rule units that enables an executable model.
```

File: drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_KIEBuilding-section.adoc [hunk 3]
```diff
@@ -313,11 +313,13 @@ In general this is not recommended, and the Maven plugin should always be used.
...
 
-*`drools-engine-classic`* is the old engine that utilizes an MVEL interpreter.  The `drools-engine-classic` and `drools-mvel` are now *deprecated*, so use `drools-engine` as possible.
+*`drools-engine`* is the standard engine for traditional DRL syntax that enables executable model.
```
Comment (kaldesai) [inline drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_KIEBuilding-section.adoc]:
```suggestion
*`drools-engine`* is the standard engine for traditional DRL syntax that enables an executable model.
```

File: drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_KIEBuilding-section.adoc [hunk 4]
```diff
@@ -313,11 +313,13 @@ In general this is not recommended, and the Maven plugin should always be used.
...
+
+*`drools-engine-classic`* is the old engine that utilizes an MVEL interpreter.  The `drools-engine-classic` and `drools-mvel` are now *deprecated*, so use `drools-ruleunits-engine` or `drools-engine` as possible.
```
Comment (kaldesai) [inline drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_KIEBuilding-section.adoc]:
```suggestion
*`drools-engine-classic`* is the old engine that utilizes an MVEL interpreter.  The `drools-engine-classic` and `drools-mvel` are now *deprecated*, so use `drools-ruleunits-engine` or `drools-engine` instead.
```

**Review by kaldesai — APPROVED** [review]
Very minor suggestions. Otherwise, your content looks good to me. Please reach out to me if you have any doubts regarding the suggestions. Thank you!

**Comment by tkobayas** [comment]
Thank you, @kaldesai !

**Comment by tkobayas** [comment]
@mariofusco Please review and merge if it's okay. Thanks!

**Review by mariofusco — APPROVED** [review]

**Pull request merged**